src/components/CalendarMonthOption.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { PropTypes } from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+import momentPropTypes from 'react-moment-proptypes';
+import moment from 'moment';
+
+export const TOUCHSTART_TIMEOUT = 200;
+
+const propTypes = {
+  month: momentPropTypes.momentObj,
+  modifiers: PropTypes.arrayOf(PropTypes.string),
+  onMonthClick: PropTypes.func,
+  onMonthMouseDown: PropTypes.func,
+  onMonthMouseUp: PropTypes.func,
+  onMonthMouseEnter: PropTypes.func,
+  onMonthMouseLeave: PropTypes.func,
+  onMonthTouchStart: PropTypes.func,
+  onMonthTouchEnd: PropTypes.func,
+  onMonthTouchTap: PropTypes.func,
+};
+
+const defaultProps = {
+  month: moment(),
+  modifiers: [],
+  onMonthClick() {},
+  onMonthMouseDown() {},
+  onMonthMouseUp() {},
+  onMonthMouseEnter() {},
+  onMonthMouseLeave() {},
+  onMonthTouchStart() {},
+  onMonthTouchEnd() {},
+  onMonthTouchTap() {},
+};
+
+export default class CalendarMonthOption extends React.Component {
+  constructor(props) {
+    super(props);
+    this.hasActiveTouchStart = false;
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState);
+  }
+
+  handleMonthClick(month, modifiers, e) {
+    this.props.onMonthClick(month, modifiers, e);
+  }
+
+  handleMonthMouseDown(month, modifiers, e) {
+    this.props.onMonthMouseDown(month, modifiers, e);
+  }
+
+  handleMonthMouseUp(month, modifiers, e) {
+    this.props.onMonthMouseUp(month, modifiers, e);
+  }
+
+  handleMonthMouseEnter(month, modifiers, e) {
+    this.props.onMonthMouseEnter(month, modifiers, e);
+  }
+
+  handleMonthMouseLeave(month, modifiers, e) {
+    this.props.onMonthMouseLeave(month, modifiers, e);
+  }
+
+  handleMonthTouchStart(month, modifiers, e) {
+    this.hasActiveTouchStart = true;
+    setTimeout(() => {
+      this.hasActiveTouchStart = false;
+    }, TOUCHSTART_TIMEOUT);
+
+    this.props.onMonthTouchStart(month, modifiers, e);
```

Comment:
please ensure every prop function is destructured, such that `this.props` doesn't end up as the `this` value of the function when invoked